### PR TITLE
Feat: deadend check

### PR DESCRIPTION
**Pull request by Tuhacrt**
Changes:
1. Added a deadend check logic to `get-available-positions.ts`
2. Added tests to `get-available-positions.test.ts` to ensure the deadend check works as intended.
3. Revised the test descriptions in `get-available-positions.test.ts` to make them more descriptive and easier to understand.

### CODE REVIEW

**Review by kayac-chang — APPROVED** [review]
LGTM

File: apps/backend/src/board/logic/get-available-positions.test.ts [hunk 1]
```diff
@@ -15,4 +38,85 @@ describe("get available positions", () => {
...
+  `, () => {
+    expect(
+      getAvailablePositions([
+        { position: [0, 0], card: PathCard.START },
+        { position: [0, -1], card: PathCard.DEADEND_TOP_LEFT_RIGHT },
+      ])
+    ).toIncludeSameMembers([
+      expect.toIncludeAllMembers([1, 0]),
+      expect.toIncludeAllMembers([0, 1]),
+      expect.toIncludeAllMembers([0, -1]),
```
Comment (nikkiching) [inline apps/backend/src/board/logic/get-available-positions.test.ts:119]: 疑問：這個test case的地圖上 [0,-1] 已經放過了，結果應該不能再出現[0,-1] 才對，可能有問題？

Comment (nikkiching) [reply apps/backend/src/board/logic/get-available-positions.test.ts:119]: @kayac-chang

Comment (Tuhacrt) [reply apps/backend/src/board/logic/get-available-positions.test.ts:119]: Yes, the expect...to algo of this test is wrong.
Can you review the updated branch and pr at 
https://github.com/Game-as-a-Service/saboteur-remix/tree/fix/get-available-positions-test
https://github.com/Game-as-a-Service/saboteur-remix/pull/52
Thank you for bug report. @nikkiching

Comment (kayac-chang) [reply apps/backend/src/board/logic/get-available-positions.test.ts:119]: @nikkiching 
對，這邊我們沒有檢查到邏輯有誤。

@Tuhacrt 
感謝修正。

Close this conversation by #52

**Pull request merged**